### PR TITLE
[JENKINS-62446] adding CasC compatibility to Global tool > Global Maven

### PR DESCRIPTION
**Pull request by aHenryJard**
See [JENKINS-62446](https://issues.jenkins-ci.org/browse/JENKINS-62446).

Global maven configuration was not exported because the symbol `@Symbol("maven")` was used twice.
- https://github.com/jenkinsci/jenkins/blob/0ce4268d219f249615ced798ed23b1cb3ecce93c/core/src/main/java/hudson/tasks/Maven.java#L427
- https://github.com/jenkinsci/jenkins/blob/0ce4268d219f249615ced798ed23b1cb3ecce93c/core/src/main/java/jenkins/mvn/GlobalMavenConfig.java#L14

I keep the `@Symbol("maven")` for the currently CasC exportable part to not break existing configuration, and renamed the one that was not working.

Manual tests done locally:
- export (see screenshot)
- import with a new configuration values on `tool.mavenGlobalConfig`

Test coverage in https://github.com/jenkinsci/configuration-as-code-plugin/pull/1525

Example:
```
tool:
  mavenGlobalConfig:
    globalSettingsProvider:
      filePath:
        path: "/tmp3/settings.xml"
    settingsProvider:
      filePath:
        path: "/tmp2/settings.xml"
```

Does anyone know if there is a JCasC compatibility test class in this project ?

FYI screenshots of UI and JCasC export:
config UI
export casc global tool

### Proposed changelog entries

* Entry 1: JCasC compatibility for Global tool (/configureTools) > Maven Configuration
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
Jcasc compat tests for core are normally hosted in the jcasc project built against an incremental of core

**Review by MRamonLeon — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]
Since the original implementation has never quite worked, I am +1 for it. At the same time, we need an explicit changelog entry to indicate what is what

**Comment by oleg-nenashev** [comment]
Arguably, the symbol collision is a bug not a feature.
@aHenryJard would you like to see it backported to the 2.263.x LTS baseline

**Comment by aHenryJard** [comment]
> Arguably, the symbol collision is a bug not a feature.
> @aHenryJard would you like to see it backported to the 2.263.x LTS baseline

Thanks for asking, no I don't have strong will to get it backported, I hit it while playing with JCasC and found it a to be good issue for my hacktober fest.

**Comment by oleg-nenashev** [comment]
@timja WDYT about backporting? It would also allow to integrate the integration test in JCasC a few months earlier

**Comment by daniel-beck** [comment]
I distinctly remember being told by someone (@jglick ?) that symbol collisions are only a problem within the same type of object, as users of symbol are expected to filter, or something along those lines. Is this a bug in JCasC?

**Comment by jglick** [comment]
Yes, sounds like a bug in JCasC, not in core.

**Review by jglick — CHANGES_REQUESTED** [review]
`@Symbol`s should _not_ include the type of extension in some phrasing. Thus `maven` was correct, `mavenGlobalConfig` incorrect. Recommend closing and solving in `configuration as-code-plugin`.

**Comment by daniel-beck** [comment]
Noting that I believe the PR description to be wrong, and this affects two different `Descriptor`s with `ToolConfigurationCategory` (the `tool` top-level container in JCasC): `Maven.MavenInstallation.DescriptorImpl` https://github.com/jenkinsci/jenkins/blob/0ce4268d219f249615ced798ed23b1cb3ecce93c/core/src/main/java/hudson/tasks/Maven.java#L665-L666 and `GlobalMavenConfig` (changed to `mavenGlobalConfig` here). That's a more awkward case for something like JCasC to handle.

**Pull request merged**
